test/models/wishlist_item_test.rb

```diff
@@ -69,7 +69,62 @@ def test_domain_name_must_be_a_valid_domain
     assert(item.valid?)
 
     item = WishlistItem.new(user: @user, domain_name: 'some')
+    assert(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'kana.')
+    assert_not(item.valid?)
+    assert_equal(['is invalid'], item.errors[:domain_name])
+  end
+
+  def test_domain_has_valid_tld
+    # This should be a generative test in the future.
+    # Valid list is populated to ee, test, pri.ee, com.ee, med.ee, fie.ee
+    item = WishlistItem.new(user: @user, domain_name: 'dupe.ee')
+    assert(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: "#{SecureRandom.hex}.pri.ee")
+    assert(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'üõöä.med.ee')
+    assert(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'foo.bar')
+    assert_not(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'foo.')
+    assert_not(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'test.pri')
+    assert_not(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'test.pri.ee.')
+    assert_not(item.valid?)
+
+    item = WishlistItem.new(user: @user, domain_name: 'kana.')
     assert_not(item.valid?)
     assert_equal(['is invalid'], item.errors[:domain_name])
   end
+
+  def test_domain_is_autocompleted_with_tld
+    # Valid list is populated to ee, test, pri.ee, com.ee, med.ee, fie.ee
```

Comment:
Would it be possible to make it self-explanatory, so that comments are not needed here? I have a difficulty understanding what is the "given" state here and what exactly is being tested.